client/signup/steps/plans/index.jsx

```diff
@@ -76,86 +70,6 @@ export class PlansStep extends Component {
 		this.unsubscribe();
 	}
 
-	onSelectPlan = ( cartItem ) => {
```

Comment:
It's unfortunate that the tests are failing @donlair, however, I think a good compromise would be to retain this function.
And call the function in step actions within this function. 

It's a bit dirty and I would rather move all those related tests to `client/lib/signup/step-actions.js` but the above fix should also work without increasing the workload even more.